core/src/repair/serve_repair.rs

```diff
@@ -122,6 +127,19 @@ impl ShredRepairType {
             | ShredRepairType::ShredForBlockId(slot, _, _) => *slot,
         }
     }
+
+    pub fn location_to_insert_response(&self) -> BlockLocation {
+        match self {
+            ShredRepairType::OrphanForBlockId(_, bid)
+            | ShredRepairType::HighestShredForBlockId(_, _, bid)
+            | ShredRepairType::ShredForBlockId(_, _, bid) => {
+                BlockLocation::Repair { block_id: *bid }
```

Comment:
maybe making a `fn block_id() -> Option<BlockId>` method on `ShredRepairType` and casing on that to determine `BlockLocation` can make it more explicit this location is based on blockid

Reply:
Was thinking something like 
```
self.block_id()
                .map(|bid| BlockLocation::Repair { block_id: *bid })
                .or_else(BlockLocation::Original);
```

Reply:
done e1cd2db0e1bcb4f98164aa3681c29267f586fa88